pkg/network/handlers/state_request.go

```diff
@@ -0,0 +1,237 @@
+package handlers
+
+import (
+	"bytes"
+	"context"
+	"crypto/ed25519"
+	"encoding/binary"
+	"fmt"
+	"github.com/eigerco/strawberry/internal/crypto"
+	"github.com/eigerco/strawberry/internal/merkle/trie"
+	"github.com/eigerco/strawberry/internal/store"
+	"github.com/eigerco/strawberry/pkg/serialization/codec/jam"
+	"github.com/quic-go/quic-go"
+)
+
+const (
+	hashSize         = crypto.HashSize
+	keySize          = 31
+	maxSizeSize      = 4
+	headerHashOffset = 0
+	keyStartOffset   = headerHashOffset + hashSize
+	keyEndOffset     = keyStartOffset + keySize
+	maxSizeOffset    = keyEndOffset + keySize
+)
+
+// StateRequestHandler processes incoming state request streams according to the CE 129 protocol.
+// It provides access to the trie data structure that stores the blockchain state.
+type StateRequestHandler struct {
+	trie *store.Trie // The state trie store containing key-value pairs
+}
+
+// NewStateRequestHandler creates a new handler for processing state requests.
+// It takes a reference to a store.Trie that will be used to serve state data.
+func NewStateRequestHandler(trie *store.Trie) *StateRequestHandler {
+	return &StateRequestHandler{
+		trie: trie,
+	}
+}
+
+// stateRequestMessage represents the structure of an incoming state request.
+type stateRequestMessage struct {
+	HeaderHash crypto.Hash   // The hash of the block header whose state is being requested
+	KeyStart   [keySize]byte // The first key in the requested range (inclusive)
```

Comment:
nit: this might be clearer as a specific type, like `TrieKey` , but it's fine like this too for now.